CMakeLists.txt

```diff
@@ -1076,6 +1087,37 @@ if (ENABLE_CGAL)
 endif()
 list(APPEND Sources src/openscad.cc ${CORE_SOURCES} ${OFFSCREEN_SOURCES})
 
+set_source_files_properties(
+  # flex and bison tends to generate many macros, which does not work well with
+  # unity build
+  ${FLEX_openscad_lexer_OUTPUTS}
+  ${BISON_openscad_parser_OUTPUTS}
+  ${FLEX_comment_lexer_OUTPUTS}
+  ${BISON_comment_parser_OUTPUTS}
+  # manifold uses a lot of anonymous namespaces.
+  # unity build will cause naming conflict for functions inside those namespaces.
+  ${MANIFOLD_SOURCES}
+  # files using opengl does not work well with unity build, again due to the
+  # macros
+  ${OFFSCREEN_SOURCES}
+  ${PLATFORM_SOURCES}
+  src/glview/OffscreenContextFactory.cc
+  src/glview/RenderSettings.cc
+  src/glview/Camera.cc
+  src/glview/ColorMap.cc
+  src/glview/preview/CSGTreeNormalizer.cc
+  src/gui/input/SpaceNavInputDriver.cc
```

Comment:
Did this file sneak in by mistake?

Reply:
no, see https://github.com/openscad/openscad/pull/4768#issuecomment-1754311120

but yeah I should document this as well